src/main/java/com/google/sps/data/OpenAQMeasurement.java

```diff
@@ -8,72 +10,26 @@
   public String unit;
   public double value;
 
-  // The thresholds for each of the AQ categories as specified at
-  // https://www.environment.nsw.gov.au/topics/air/understanding-air-quality-data/air-quality-categories
-  // Some values have been rounded down to allow for integer storage
-  private int[] o3Thresholds = {6, 10, 15, 20}; // pphm
-  private int[] no2Thresholds = {8, 12, 18, 24}; // pphm
-  private int[] so2Thresholds = {13, 20, 30, 40}; // pphm
-  private int[] coThresholds = {6, 9, 15, 18}; // ppm
-  private int[] pm10Thresholds = {50, 100, 200, 600}; // µg/m³
-  private int[] pm25Thresholds = {25, 50, 100, 300}; // µg/m³
-
-  /*
-  * Calculates the AQ category as specified by
-  * https://www.environment.nsw.gov.au/topics/air/understanding-air-quality-data/air-quality-categories
-  * Note: the category names differ from above, instead the names from the following site are used:
-  * https://soe.environment.gov.au/theme/ambient-air-quality/topic/2016/air-quality-index
-  * The category is returned as a number 1-5 or -1:
-  * -1: The either the conversion for the given unit or the parameter is not supported
-  * 1: Very Good
-  * 2: Good
-  * 3: Fair
-  * 4: Poor
-  * 5: Very Poor
-  */
-  public int calcAQCategory() {
-    switch (this.name) {
-      case "o3":
-        if (this.unit.equals("ppm")) {
-          return getCategoryFromThresholds(this.value * 100, o3Thresholds);
-        }
-        return -1;
-      case "no2":
-        if (this.unit.equals("ppm")) {
-          return getCategoryFromThresholds(this.value * 100, no2Thresholds);
-        }
-        return -1;
-      case "so2":
-        if (this.unit.equals("ppm")) {
-          return getCategoryFromThresholds(this.value * 100, so2Thresholds);
-        }
-        return -1;
-      case "co":
-        if (this.unit.equals("ppm")) {
-          return getCategoryFromThresholds(this.value, coThresholds);
-        }
-        return -1;
-      case "pm10":
-        if (this.unit.equals("µg/m³")) {
-          return getCategoryFromThresholds(this.value, pm10Thresholds);
-        }
-        return -1;
-      case "pm25":
-        if (this.unit.equals("µg/m³")) {
-          return getCategoryFromThresholds(this.value, pm25Thresholds);
-        }
-        return -1;
-      }
+  private HashMap<String, Double> australianMaxStandards = createStandardsMap();
 
-    return -1;
+  private HashMap<String, Double> createStandardsMap() {
+    HashMap<String, Double> map = new HashMap<>();
+    map.put("o3_ppm", 0.1);
+    map.put("no2_ppm", 0.12);
+    map.put("so2_ppm", 0.2);
+    map.put("co_ppm", 9.0);
+    map.put("pm10_µg/m³", 50.0);
+    map.put("pm25_µg/m³", 25.0);
+    return map;
   }
 
```

Comment:
what are the advantages of refactoring it into a map over what was there before?

Reply:
It's easier to interpret than a giant switch statement.

Also, this was a whole new way of calculating the AQI that required rewriting code, and last time I didn't think of using a map with `map.getOrDefault` to only use measurements that I could turn into an AQI

Reply:
To clarify the purpose of this PR a bit, in the old version I would calculate the AQ category, then the AQI would be the midpoint of the category's threshold, which wasn't very useful or interesting data wise.

Now, it calculates the AQI for each individual measurement (which I didn't realise was a thing before) (i.e. pm10, pm25, co, no2) then works out the maximum for each station, no need for categories.

Reply:
ah, that makes more sense, thanks!